settings.py

```diff
@@ -315,36 +317,35 @@
-    SOCIAL_AUTH_GOOGLE_OAUTH2_AUTH_EXTRA_ARGUMENTS = {
-        'access_type': 'offline',  # to make the access_token can be refreshed after expired (expiration time is 1 hour)
-    }
+# Use Google sub ID as the user ID, safer than using email
+USE_UNIQUE_USER_ID = True
+
+SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = GOOGLE_AUTH_CLIENT_CONFIG.get('web', {}).get('client_id')
+SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET = GOOGLE_AUTH_CLIENT_CONFIG.get('web', {}).get('client_secret')
 
-    # Use Google sub ID as the user ID, safer than using email
-    USE_UNIQUE_USER_ID = True
+SOCIAL_AUTH_POSTGRES_JSONFIELD = True
+SOCIAL_AUTH_URL_NAMESPACE = 'social'
+SOCIAL_AUTH_LOGIN_REDIRECT_URL = '/'
 
-    SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = GOOGLE_AUTH_CLIENT_CONFIG.get('web', {}).get('client_id')
-    SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET = GOOGLE_AUTH_CLIENT_CONFIG.get('web', {}).get('client_secret')
+SOCIAL_AUTH_PIPELINE = (
+    'social_core.pipeline.social_auth.social_details',
+    'social_core.pipeline.social_auth.social_uid',
+    'social_core.pipeline.social_auth.social_user',
+    'social_core.pipeline.social_auth.associate_by_email',
+    'social_core.pipeline.social_auth.associate_user',
+    'seqr.utils.social_auth_pipeline.log_signed_in',
+)
 
-    SOCIAL_AUTH_GOOGLE_PLUS_AUTH_EXTRA_ARGUMENTS = {
-          'access_type': 'offline'
+if TERRA_API_ROOT_URL:
+    SOCIAL_AUTH_GOOGLE_OAUTH2_AUTH_EXTRA_ARGUMENTS = {
+        'access_type': 'offline',  # to make the access_token can be refreshed after expired (expiration time is 1 hour)
     }
 
-    SOCIAL_AUTH_POSTGRES_JSONFIELD = True
-    SOCIAL_AUTH_URL_NAMESPACE = 'social'
-    SOCIAL_AUTH_LOGIN_REDIRECT_URL = '/'
-    SOCIAL_AUTH_PIPELINE = (
-        'seqr.utils.social_auth_pipeline.validate_anvil_registration',
-        'social_core.pipeline.social_auth.social_details',
-        'social_core.pipeline.social_auth.social_uid',
-        'social_core.pipeline.social_auth.social_user',
-        'social_core.pipeline.user.get_username',
-        'social_core.pipeline.social_auth.associate_by_email',
-        'social_core.pipeline.user.create_user',
-        'social_core.pipeline.social_auth.associate_user',
-        'social_core.pipeline.social_auth.load_extra_data',
-        'social_core.pipeline.user.user_details',
-        'seqr.utils.social_auth_pipeline.log_signed_in',
-    )
-    INSTALLED_APPS.append('social_django')
+    SOCIAL_AUTH_PIPELINE = ('seqr.utils.social_auth_pipeline.validate_anvil_registration',) + \
+                           SOCIAL_AUTH_PIPELINE[:4] + \
+                           ('social_core.pipeline.user.get_username',
+                           'social_core.pipeline.user.create_user') + \
+                           SOCIAL_AUTH_PIPELINE[4:5] + \
+                           ('social_core.pipeline.social_auth.load_extra_data',
+                           'social_core.pipeline.user.user_details') + \
+                           SOCIAL_AUTH_PIPELINE[5:]
```

Comment:
I know this was my suggestion, but seeing it its still hard to read. Maybe do something like

```
SOCIAL_AUTH_INITIAL_PIPELINE = (
    'social_core.pipeline.social_auth.social_details',
    'social_core.pipeline.social_auth.social_uid',
    'social_core.pipeline.social_auth.social_user',
    'social_core.pipeline.social_auth.associate_by_email',
    'seqr.utils.social_auth_pipeline.log_signed_in',
)
SOCIAL_AUTH_ASSOCIATE_USER_PIPELINE = ( 'social_core.pipeline.social_auth.associate_user',)
SOCIAL_AUTH_LOG_PIPELINE = ( 'seqr.utils.social_auth_pipeline.log_signed_in',)
...
if TERRA_API_ROOT_URL:
   ...
   SOCIAL_AUTH_PIPELINE = ('seqr.utils.social_auth_pipeline.validate_anvil_registration',) + \
                           SOCIAL_AUTH_INITIAL_PIPELINE + \
                           ('social_core.pipeline.user.get_username',
                           'social_core.pipeline.user.create_user') + \
                           SOCIAL_AUTH_ASSOCIATE_USER_PIPELINE + \
                           ('social_core.pipeline.social_auth.load_extra_data',
                           'social_core.pipeline.user.user_details') + \
                           SOCIAL_AUTH_LOG_PIPELINE
else:
   SOCIAL_AUTH_PIPELINE = SOCIAL_AUTH_INITIAL_PIPELINE + SOCIAL_AUTH_ASSOCIATE_USER_PIPELINE + SOCIAL_AUTH_LOG_PIPELINE
```